dlt/destinations/sql_jobs.py

```diff
@@ -333,6 +337,44 @@ def gen_merge_sql(
                 )
             )
 
+        # remove "non-latest" records from staging table (deduplicate) if a sort column is provided
+        if len(primary_keys) > 0:
+            if has_column_with_prop(root_table, "sort"):
+                sort_column = escape_identifier(get_columns_names_with_prop(root_table, "sort")[0])
+                sql.append(f"""
+                    DELETE FROM {staging_root_table_name}
+                    WHERE {sort_column} IN (
+                        SELECT {sort_column} FROM (
+                            SELECT {sort_column}, ROW_NUMBER() OVER (partition BY {", ".join(primary_keys)} ORDER BY {sort_column} DESC) AS _rn
+                            FROM {staging_root_table_name}
+                        ) AS a
+                        WHERE a._rn > 1
+                    );
+                """)
+
+        # remove deleted records from staging tables if a hard_delete column is provided
+        if has_column_with_prop(root_table, "hard_delete"):
+            hard_delete_column = escape_identifier(
+                get_columns_names_with_prop(root_table, "hard_delete")[0]
+            )
+            # first delete from root staging table
+            sql.append(f"""
+                DELETE FROM {staging_root_table_name}
+                WHERE {hard_delete_column} IS NOT DISTINCT FROM {escape_literal(True)};
```

Comment:
ok so you assume that hard deleted column is boolean. probably makes the most sense. but then you must check the type somewhere. my take:
delete if value is IS NOT NULL OR (only in case of boolean) when true as above. maybe someone wants to have deleted flag as timestamp?

Reply:
Good point, I implemented your suggestion.